node/src/components/storage.rs

```diff
@@ -958,6 +964,11 @@ impl Storage {
                     } else {
                         return Ok(responder.respond(None).ignore());
                     };
+
+                if !self.should_return_block(block.height(), only_from_highest_contiguous_range) {
+                    return Ok(responder.respond(None).ignore());
+                }
```

Comment:
We could do this even before we create a txn or call `get_single_block`. In the other place as well.